### PR TITLE
Blogroll: create blocks from recommendations

### PR DESCRIPTION
**Pull request by heavyweight**
This PR creates the inner blocks ( blogroll items ) based on the response received by the recommendations endpoint.

p9Jlb4-8nF-p2#adding-the-block

Fixes #
https://github.com/Automattic/wp-calypso/issues/81351

## Proposed changes:
* Fetch recommendations and use them to create inner blocks
* Create inner template for blogroll item
* Styling fixed 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to  the editor
* Add the blogroll block
* Ensure that the response from `recommendations` is used to create the initial content.

### CODE REVIEW

**Review by alshakero — APPROVED** [review]
Looks and works great. I found two issues:

1. Some sites don't have titles. We should use the URL in that case.
2. if the site is in dark mode, the item block is broken.
image


Mergable nonetheless.

**Comment by agrullon95** [comment]
@alshakero -

Im not seeing the same output in my editor: 
The placeholder image is overflowing and not properly setting to 48px width/height.

Are you seeing this?
CleanShot 2023-09-06 at 15 11 25@2x

Im also not seeing anything related recommendations endpoint from the PR description: `This PR creates the inner blocks ( blogroll items ) based on the response received by the recommendations endpoint.`. Im assuming this description has been mistakenly added, if thats the case, then im good to merge on my end.

File: projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/editor.scss [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+	font-size: 14px;
+	font-style: normal;
+	font-weight: 400;
+
+	.wp-block-columns {
+		padding: 20px 8px;
+		margin: 0;
+	}
+
+}
```
Comment (agrullon95) [inline projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/editor.scss:12]: If someone else can confirm they're also seeing the blockroll item icon placeholder image overflowing in the editor, we can add this styling to set it to 48px width/height:

CleanShot 2023-09-06 at 16 19 17@2x


```
.components-placeholder.block-editor-media-placeholder {
	min-width: 48px;
    	min-height: 48px;
    	height: 48px;
}
```

**Comment by alshakero** [comment]
Maybe I broke something when I fixed the conflicts. I'll check.

**Review by agrullon95 — APPROVED** [review]

**Pull request merged**